From af99a52d41cf800f9227f46d6ef9d992fb54dd68 Mon Sep 17 00:00:00 2001
From: "Richard A. Griffiths" <richard.a.griffiths@intel.com>
Date: Wed, 2 Sep 2015 22:18:00 -0700
Subject: [PATCH 12/17] input:silead: defer probing reduces impact of fw load

The Silead touchscreen loads it's firmware over i2c which
takes ~4s. By deferring the driver probe the startup performance
impact is reduced as measured by cold_boot_time_info.py.

Change-Id: Iae34ee99619b3ebc35ce053a4c5de8144103b6a8
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-20919
Signed-off-by: Richard A. Griffiths <richard.a.griffiths@intel.com>
Reviewed-on: https://android.intel.com:443/472785
---
 drivers/input/touchscreen/silead.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/input/touchscreen/silead.c b/drivers/input/touchscreen/silead.c
index 2517d0e..dc8a352 100644
--- a/drivers/input/touchscreen/silead.c
+++ b/drivers/input/touchscreen/silead.c
@@ -562,6 +562,12 @@ static int silead_ts_probe(struct i2c_client *client,
 	struct silead_ts_data *data;
 	struct device *dev = &client->dev;
 	int ret;
+	static int first_time = 0;
+/* loading the firmware to the Silead takes about 4 seconds, so let's defer
+ * probing until we are multi-threaded
+ */
+	if(++first_time == 1)
+		return -EPROBE_DEFER;
 
 	if (!i2c_check_functionality(client->adapter,
 				     I2C_FUNC_I2C |
-- 
1.9.1

